File: Mobiles/flutter/lib/core/application_layer/o11y/metrics/o11y_metrics.dart
import 'package:faro/faro.dart';
import 'package:flutter_mobile_o11y_demo/core/application_layer/o11y/faro/faro.dart';

class O11yMetrics {
  O11yMetrics() : _faro = faro;

  final Faro _faro;

  void addMeasurement(String name, Map<String, dynamic> values) {
    _faro.pushMeasurement(values, name);
  }
}

final o11yMetrics = O11yMetrics();

File: Mobiles/react-native/src/core/o11y/o11yMetrics.ts
import { faro } from '@grafana/faro-react-native';

export function addMeasurement(
  name: string,
  values: Record<string, number>,
): void {
  if (!faro?.api) return;

  faro.api.pushMeasurement({
    type: name,
    values,
  });
}
